ulib/libax/src/cbindings/io_mpx/epoll.rs

```diff
@@ -0,0 +1,280 @@
+use crate::cbindings::{
+    ctypes,
+    fd_ops::{add_file_like, get_file_like, FileLike},
+};
+use crate::debug;
+use crate::sync::Mutex;
+use axerrno::{LinuxError, LinuxResult};
+
+use alloc::collections::{BTreeMap, BTreeSet};
+use alloc::sync::Arc;
+use core::ffi::c_int;
+
+pub struct EPollCtlOp;
+impl EPollCtlOp {
+    const ADD: i32 = 1; /* Add a file descriptor to the interface.  */
+    const DEL: i32 = 2; /* Remove a file descriptor from the interface.  */
+    const MOD: i32 = 3; /* Change file descriptor epoll_event structure.  */
+}
+
+#[derive(Clone, Copy)]
+pub struct EpollData {
```

Comment:
Use `ctypes::epoll_data` and `types::epoll_event`.

Reply:
There is one problem:
`EpollInstance` contains `epoll_data` and `epoll_event`, and has to be `Send` and `Sync`. 
However `ctypes::epoll_data` contains C raw pointer `*mut c_void` which is not `Send` and `Sync`.
Here I force `unsafe impl Send/Sync for EpollInstance` to reach this goal. Is this acceptable? Or change `ctypes::epoll_data` into u64 only, without raw pointer?

Reply:
The current one is OK.